core/src/main/java/org/ehcache/Ehcache.java

```diff
@@ -245,6 +248,7 @@ public V apply(final K key, final V previousValue) {
       ValueHolder<V> computed = store.compute(key, remappingFunction);
       if (computed != null) {
         putObserver.end(PutOutcome.ADDED);
+        eventNotificationService.fireEvents();
```

Comment:
Why doing this only in one branch? What happens otherwise if expiry or evictions were triggered by the logic?